.circleci/config.yml

```diff
@@ -1,42 +1,118 @@
 version: 2.1
 
-jobs:
-  standards:
+commands:
+  install-dependencies:
+    description: "Install development dependencies."
+    steps:
+      - run: composer install
+      - run: npm ci
+  mkdir_artifacts:
+    description: "Make Artifacts directory"
+    steps:
+      - run:
+          command: |
+            [ ! -d "/tmp/artifacts" ] && mkdir /tmp/artifacts &>/dev/null
+
+executors:
+  base:
+    docker:
+      - image: circleci/buildpack-deps:latest
+    working_directory: /tmp
+  php_node:
     docker:
       - image: circleci/php:7.3.3-stretch-node-browsers
+    working_directory: /tmp/genesis-sample
+  python:
+    docker:
+      - image: circleci/python:3.7-stretch
+    working_directory: /tmp
+
+jobs:
+  checkout:
+    executor: base
     steps:
-      - checkout
+      - checkout:
+          path: genesis-sample
+      - persist_to_workspace:
+          root: /tmp
+          paths:
+            - genesis-sample
+
+  standards:
+    executor: php_node
+    steps:
+      - attach_workspace:
+          at: /tmp
       - install-dependencies
       - run: composer phpcs
       - run: npm run lint:css
       - run: npm run lint:js
 
   bundle:
-    docker:
-      - image: circleci/node:12.12.0
+    executor: php_node
     steps:
-      - checkout
+      - attach_workspace:
+          at: /tmp
       - run: npm ci
       - run: npm run zip
-      - run: mkdir /tmp/artifacts
+      - mkdir_artifacts
       - run: mv ./*.zip /tmp/artifacts/.
+      - persist_to_workspace:
+          root: /tmp
+          paths:
+            - artifacts
+            - genesis-sample
       - store_artifacts:
           path: /tmp/artifacts/
 
-commands:
-  install-dependencies:
-    description: "Install development dependencies."
+  create_data_file:
+    executor: php_node
     steps:
-      - run: composer install
-      - run: npm ci
+      - attach_workspace:
+          at: /tmp
+      - run: ls -lash
+
+  deploy_s3:
+    executor: python
+    steps:
+      - attach_workspace:
+          at: /tmp
+      - aws-s3/sync:
+          from: "/tmp/artifacts/"
+          to: "s3://<need_bucket_url>"
+
 workflows:
   version: 2
   check-standards:
     jobs:
-      - standards
+      - checkout
+      - standards:
+          requires:
+            - checkout
       - bundle:
+          requires:
+            - standards
+          filters:
+            tags:
+              only: /.*/
+            branches:
+              only: master
```

Comment:
Any reason to build the zip bundle only on master and not on every PR?  Having that zip there to download is a way someone on the team could download what was built and test it.

Reply:
Noted for the next PR I'll be making